packages/api/src/routes/user.ts

```diff
@@ -172,9 +179,10 @@ async function revokeUserProviderAccess(
     const token = connectedUser.dataValues.providerMap?.garmin?.token;
     const cxId = connectedUser.dataValues.cxId;
     if (token) await Constants.PROVIDER_OAUTH1_MAP[providerOAuth1.data].deregister([token], cxId);
-  } else if (provider === PROVIDER_APPLE) {
-    const apple = new Apple();
-    await apple.revokeProviderAccess(connectedUser);
+  } else if (Object.keys(noAuthProviders).includes(provider)) {
+    const prov = provider as typeof PROVIDER_APPLE | typeof PROVIDER_TENOVI;
+    const noAuthProvider = new (noAuthProviders[prov] as typeof Apple | typeof Tenovi)();
```

Comment:
I suggest we keep the same approach we have for the conditionals on line 181 and 183. Just need to add Tenovi to the no-auth schema on `constants.ts`:

```ts
export const providerNoAuthSchema = z.enum([PROVIDER_APPLE, PROVIDER_TENOVI]);
```

...and update the code here to look like the other scenarios on the "thread" of conditionals.

The key here is to have strong types, minimizing the usage of `as` which removes compiler verification.